VPLAY-8920  AAMP TSB - Progress reporting

anjali-syna opened the pull request:
There are no L2 tests to verify ProgressReports in  Lowest and highest trickmode speeds(x2 and x64) and normal play and pause. Adding them as part of VPLAY-8920. Adding a log in aamp cli, so that l2 tests can use it for verifying tsb duration.

anjali-syna committed:
VPLAY-8920 [ENT-4082] AAMP TSB - Progress reporting

Reason for change: There are no L2 tests to verify ProgressReports in
 Lowest and highest trickmode speeds(x2 and x64) and normal play and pause.

Changes:
* Adding L2 tests to verify ProgressReport in lowest and highest trickmode
  speeds(x2 and x64) as well as normal play and pause.

Test Procedure: Run VPLAY-8920 L2 tests 5001 and 5002

Risks: Low

anjali-syna committed:
VPLAY-8920 [ENT-4082] AAMP TSB - Progress reporting

Reason for change: There are no L2 tests to verify ProgressReports in
 Lowest and highest trickmode speeds(x2 and x64) and normal play and pause.

Changes:
* Adding L2 tests to verify ProgressReport in lowest and highest trickmode
  speeds(x2 and x64) as well as normal play and pause.

Test Procedure: Run VPLAY-8920 L2 tests 5001 and 5002

Risks: Low

anjali-syna committed:
VPLAY-8920 [ENT-4082] AAMP TSB - Progress reporting

Reason for change: There are no L2 tests to verify ProgressReports in
 Lowest and highest trickmode speeds(x2 and x64) and normal play and pause.

Changes:
* Adding L2 tests to verify ProgressReport in lowest and highest trickmode
  speeds(x2 and x64) as well as normal play and pause.

Test Procedure: Run VPLAY-8920 L2 tests 5001 and 5002

Risks: Low

anjali-syna committed:
VPLAY-8920 [ENT-4082] AAMP TSB - Progress reporting

Reason for change: There are no L2 tests to verify ProgressReports in
 Lowest and highest trickmode speeds(x2 and x64) and normal play and pause.

Changes:
* Adding L2 tests to verify ProgressReport in lowest and highest trickmode
  speeds(x2 and x64) as well as normal play and pause.

Test Procedure: Run VPLAY-8920 L2 tests 5001 and 5002

Risks: Low

Rebasing to latest changes in dev_sprint_25_2

jfagunde reviewed (approved):
Looks good to me, +1

anjali-syna committed:
VPLAY-8920 [ENT-4082] AAMP TSB - Progress reporting

Reason for change: There are no L2 tests to verify ProgressReports in
 Lowest and highest trickmode speeds(x2 and x64) and normal play and pause.

Changes:
* Adding L2 tests to verify ProgressReport in lowest and highest trickmode
  speeds(x2 and x64) as well as normal play and pause. The progress report
  log in aampcli is unable to print the seekableRange completely when
  absolute time is used. Fixing this bug in aampcli and adding new command
  to print the progress report log in single line.

Test Procedure: Run L2 tests 5001 and 5002

Risks: Low

Signed-off-by: anjali-syna <206662904+anjali-syna@users.noreply.github.com>

anjali-syna committed:
VPLAY-8920 [ENT-4082] AAMP TSB - Progress reporting

Reason for change: There are no L2 tests to verify ProgressReports in
 Lowest and highest trickmode speeds(x2 and x64) and normal play and pause.

Changes:
* Adding L2 tests to verify ProgressReport in lowest and highest trickmode
  speeds(x2 and x64) as well as normal play and pause. The progress report
  log in aampcli is unable to print the seekableRange completely when
  absolute time is used. Fixing this bug in aampcli and adding new command
  to print the progress report log in single line.

Test Procedure: Run L2 tests 5001 and 5002

Risks: Low

Signed-off-by: anjali-syna <206662904+anjali-syna@users.noreply.github.com>

anjali-syna committed:
VPLAY-8920 [ENT-4082] AAMP TSB - Progress reporting

Reason for change: There are no L2 tests to verify ProgressReports in
 Lowest and highest trickmode speeds(x2 and x64) and normal play and pause.

Changes:
* Adding L2 tests to verify ProgressReport in lowest and highest trickmode
  speeds(x2 and x64) as well as normal play and pause. The progress report
  log in aampcli is unable to print the seekableRange completely when
  absolute time is used. Fixing this bug in aampcli and adding new command
  to print the progress report log in single line.

Test Procedure: Run L2 tests 5001 and 5002

Risks: Low

Signed-off-by: anjali-syna <206662904+anjali-syna@users.noreply.github.com>

anjali-syna committed:
VPLAY-8920 [ENT-4082] AAMP TSB - Progress reporting

Reason for change: There are no L2 tests to verify ProgressReports in
 Lowest and highest trickmode speeds(x2 and x64) and normal play and pause.

Changes:
* Adding L2 tests to verify ProgressReport in lowest and highest trickmode
  speeds(x2 and x64) as well as normal play and pause. The progress report
  log in aampcli is unable to print the seekableRange completely when
  absolute time is used. Fixing this bug in aampcli and adding new command
  to print the progress report log in single line.

Test Procedure: Run L2 tests 5001 and 5002

Risks: Low

Signed-off-by: anjali-syna <206662904+anjali-syna@users.noreply.github.com>

jfagunde reviewed (approved):
+1

cpc005 reviewed (approved):
You might have to put in new license text.

anjali-syna committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-8920

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-8920

anjali-syna committed:
VPLAY-8920 [ENT-4082] AAMP TSB - Progress reporting

Reason for change: There are no L2 tests to verify ProgressReports in
 Lowest and highest trickmode speeds(x2 and x64) and normal play and pause.

Changes:
* Adding L2 tests to verify ProgressReport in lowest and highest trickmode
  speeds(x2 and x64) as well as normal play and pause. The progress report
  log in aampcli is unable to print the seekableRange completely when
  absolute time is used. Fixing this bug in aampcli and adding new command
  to print the progress report log in single line.

Test Procedure: Run L2 tests 5001 and 5002

Risks: Low

Signed-off-by: anjali-syna <206662904+anjali-syna@users.noreply.github.com>

anjali-syna committed:
VPLAY-8920 [ENT-4082] AAMP TSB - Progress reporting

Reason for change: There are no L2 tests to verify ProgressReports in
 Lowest and highest trickmode speeds(x2 and x64) and normal play and pause.

Changes:
* Adding L2 tests to verify ProgressReport in lowest and highest trickmode
  speeds(x2 and x64) as well as normal play and pause. The progress report
  log in aampcli is unable to print the seekableRange completely when
  absolute time is used. Fixing this bug in aampcli and adding new command
  to print the progress report log in single line.

Test Procedure: Run L2 tests 5001 and 5002

Risks: Low

Signed-off-by: anjali-syna <206662904+anjali-syna@users.noreply.github.com>

anjali-syna committed:
Pulling the latest from remote.
Merge branch 'feature/VPLAY-8920' of github.com:rdkcentral/aamp into feature/VPLAY-8920

jfagunde committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-8920

anjali-syna committed:
VPLAY-8920 [ENT-4082] AAMP TSB - Progress reporting

Reason for change: There are no L2 tests to verify ProgressReports in
 Lowest and highest trickmode speeds(x2 and x64) and normal play and pause.

Changes:
* Adding L2 tests to verify ProgressReport in lowest and highest trickmode
  speeds(x2 and x64) as well as normal play and pause. The progress report
  log in aampcli is unable to print the seekableRange completely when
  absolute time is used. Fixing this bug in aampcli and changing the
  progress report log to single line.

Test Procedure: Run L2 tests 5001 and 5002

Risks: Low

Merge branch 'feature/VPLAY-8920' of github.com:rdkcentral/aamp into feature/VPLAY-8920

anjali-syna committed:
VPLAY-8920 [ENT-4082] AAMP TSB - Progress reporting

Reason for change: There are no L2 tests to verify ProgressReports in
 Lowest and highest trickmode speeds(x2 and x64) and normal play and pause.

Changes:
* Adding L2 tests to verify ProgressReport in lowest and highest trickmode
  speeds(x2 and x64) as well as normal play and pause. The progress report
  log in aampcli is unable to print the seekableRange completely when
  absolute time is used. Fixing this bug in aampcli and changing the
  progress report log to single line.

Test Procedure: Run L2 tests 5001 and 5002

Risks: Low

Signed-off-by: anjali-syna <206662904+anjali-syna@users.noreply.github.com>

anjali-syna committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-8920

anjali-syna committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-8920

anjali-syna commented on test/aampcli/Aampcli.cpp:
```diff
@@ -530,6 +530,7 @@ void MyAAMPEventListener::Event(const AAMPEventPtr& e)
 		case AAMP_EVENT_PROGRESS:
 			{
 				ProgressEventPtr ev = std::dynamic_pointer_cast<ProgressEvent>(e);
+				printf("[AAMLCLI] AAMP_EVENT_PROGRESS: start:%ldms, position:%ldms, end:%ldms, duration:%ldms, rate:%.2f\n", (long)(ev->getStart()), (long)(ev->getPosition()), (long) (ev->getEnd()), (long) (ev->getDuration()), ev->getCurrentPlayRate());
```
It is already corrected in the newer commits of this PR :-)
